processing/src/main/java/org/apache/druid/java/util/common/granularity/Granularity.java

```diff
@@ -290,13 +288,22 @@ public boolean hasNext()
     @Override
     public Interval next()
     {
+      if (limit <= 0) {
+        throw new IAE(
+            "Limit [%s] exceeded while iterating intervals for granularity [%s] and input interval [%s]",
+            limit,
```

Comment:
Given that you are decrementing `limit`, the value interpolated into this log is not actually going to be that useful...